form/form.go

```diff
@@ -0,0 +1,223 @@
+// Copyright 2015 Canonical Ltd.
+// Licensed under the LGPLv3, see LICENCE file for details.
+
+//Package form enables the completion of forms specified as using environschema.
+package form
+
+import (
+	"fmt"
+	"io"
+	"os"
+	"sort"
+	"strings"
+
+	"github.com/juju/schema"
+	"golang.org/x/crypto/ssh/terminal"
+	"gopkg.in/errgo.v1"
+	"gopkg.in/juju/environschema.v1"
+)
+
+// PromptingFiller fills out a form described by an environscheama by
```

Comment:
// PromptingFiller fills out a set of environschema fields by
// prompting the user for each field in sequence.

?